src/features/bumpkins/actions/buildNPCSheets.ts

```diff
@@ -44,39 +43,55 @@ const URL =
     ? "https://images.bumpkins.io/npcSheets"
     : "https://testnet-images.bumpkins.io/npcSheets";
 
-export async function buildNPCSheets(request: Request): Promise<Response> {
-  const tokenUri = tokenUriBuilder(request.parts);
-  const response: Response = { sheets: {} as Sheets };
+const getImage = async (url: string) => {
+  let image: string;
 
-  await Promise.all(
-    SHEET_TYPES.map(async (sheetType) => {
-      const url = `${URL}/${sheetType}/${tokenUri}.webp`;
+  try {
+    image = await new Promise((resolve, reject) => {
       const img = new Image();
+      img.crossOrigin = "anonymous";
```

Comment:
This is important so that the XHR request sends CORS headers.
Otherwise it seems we might get an incorrect cached image in our disk cache. Not sure exactly why that happens.